### PR TITLE
fix aggregate math query

### PR DESCRIPTION
**Pull request by mengjinglei**
pr #4984 fix math query without aggregate functions, this pr makes math query with aggregate functions
 works

- [ ] CHANGELOG.md updated
- [x] Rebased/mergable
- [x] Tests pass
- [x] Sign [CLA](http://influxdb.com/community/cla.html) (if not already signed)

### CODE REVIEW

File: tsdb/aggregate.go [hunk 1]
```diff
@@ -381,8 +381,18 @@ func (e *AggregateExecutor) processFunctions(results [][]interface{}, columnName
 	callInPosition := e.stmt.FunctionCallsByPosition()
 	hasTimeField := e.stmt.HasTimeFieldSpecified()
 
+	FlatCallInPositions := make([][]*influxql.Call, 0)
```
Comment (corylanou) [inline tsdb/aggregate.go]: Comment here might help clarify for the next person why you need this.  I understand it, but I suspect this is confusing to people seeing it for the first time.

File: tsdb/aggregate.go [hunk 2]
```diff
@@ -412,8 +422,10 @@ func (e *AggregateExecutor) processFunctions(results [][]interface{}, columnName
 func (e *AggregateExecutor) processSelectors(results [][]interface{}, callPosition int, hasTimeField bool, columnNames []string) ([][]interface{}, error) {
 	// if the columns doesn't have enough columns, expand it
```
Comment (corylanou) [inline tsdb/aggregate.go]: This comment should get changed to explain why we expand or contract.

**Comment by corylanou** [comment]
Couple of minor nits.  Code looks good.  

Would be nice to have some integration tests, but we can add those later (and get cluster tests for free when that gets merged).

Thanks! +1

File: tsdb/aggregate.go (hunk 1 above)
Comment (dgnorton) [inline tsdb/aggregate.go]: also, should be `flatCallInPositions`

**Comment by dgnorton** [comment]
One very small nit.  Otherwise, +1

Thanks!

**Comment by otoolep** [comment]
We may need to merge as is to meet the 0.9.6 release deadline, and then fix it up ourselves. Not sure when @mengjinglei will get to this, and the CLA has been signed.

OK @dgnorton @corylanou ?

**Comment by otoolep** [comment]
I can fix up post-merge.

**Pull request merged**